api/src/main/java/org/apache/iceberg/SnapshotUpdate.java

```diff
@@ -52,4 +53,13 @@
    */
   ThisT stageOnly();
 
+  /**
+   * Use a particular executor to plan. The default worker pool will be used by default.
+   *
+   * @param executorService the provided executor.
+   * @return this for method chaining.
+   */
+  default ThisT planWith(ExecutorService executorService) {
```

Comment:
I wouldn't consider this planning for snapshot updates. I think a better configuration property is `scanManifestsWith`, closer to the original changes in #4115.